src/cirrus/cirrus_setup.py

```diff
@@ -216,32 +216,54 @@ def build_parser(argslist):
         help='Credential plugin manager to use',
         default=None,
     )
+    parser.add_argument(
+        '--robot', '-r',
+        dest='robot_mode',
+        action='store_true',
+        help='Non interactive robot mode installer',
+        default=False
+    )
+    parser.add_argument(
+        '--pypi-username',
+        default=None,
+        help='pypi username'
+    )
+    parser.add_argument(
+        '--pypi-token',
+        default=None,
+        help='pypi access token'
+    )
+    parser.add_argument(
+        '--github-token',
+        default=None,
+        help='github access token'
+    )
+    parser.add_argument(
+        '--github-username',
+        default=None,
+        help='github username'
+    )
+    parser.add_argument(
+        '--ssh-username',
+        default=None,
+        help='ssh username'
+    )
+    parser.add_argument(
+        '--ssh-keyfile',
+        default=None,
+        help='ssh keyfile path'
+    )
+
     opts = parser.parse_args(argslist)
     return opts
 
 
-def main():
+def interactive_setup(opts, config):
     """
-    _main_
+    interactive Q&A style config
 
-    Execute selfsetup command
     """
-    opts = build_parser(sys.argv[1:])
-    config = load_setup_configuration()
-
-    # make sure gitconfig has a cirrus section
-    if 'cirrus' not in config.gitconfig.sections:
-        config.gitconfig.add_section('cirrus')
-    config.gitconfig.set_param(
-        'alias',
-        'cirrus',
-        '! {0}/bin/cirrus'.format(os.environ['VIRTUALENV_HOME'])
-    )
-
-    # make sure the creds plugin value is set
-    if opts.cred_plugin is not None:
-        config._set_creds_plugin(opts.cred_plugin)
-
+    #TODO: Allow cli options if provided instead of prompting
```

Comment:
Seems like it would make sense to unify the robot and interactive methods a bit. The behavior I would find natural would be to read the existing .gitconfig, then call a function to replace values with cli options, then if not robot mode do the interactive prompting.